Added documentation to use Delta Live Tables migration (#3587)
databrickslabs/ucx: docs/ucx/docs/process/index.mdx

3. [table migration](/docs/process/#table-migration-process)
4. [data reconciliation](/docs/reference/workflows#post-migration-data-reconciliation-workflow)
5. [code migration](/docs/reference/commands#code-migration-commands)

> 6. [pipeline migration](/docs/process#pipeline-migration-process)

Reviewer: ```suggestion
6. [delta live table pipeline migration](/docs/process#pipeline-migration-process)
```

The [`revert-migrated-tables` command](/docs/reference/commands#revert-migrated-tables) drops the Unity Catalog table or view and reset
the `upgraded_to` property on the source object. Use this command to allow for migrating a table or view again.

## Pipeline Migration Process

> You are required to complete the [assessment workflow](/docs/reference/workflows#assessment-workflow) before starting the pipeline migration workflow.

> The pipeline migration process is an automated process that migrates Hive Metastore Delta Live Table (DLT) pipelines to the Unity Catalog.

Reviewer: What is the correct wording here? I think, it updates pipelines (in place) to update tables in UC instead of Hive metastore, right?

```suggestion
The pipeline migration process is a workflow that updates Delta Live Table (DLT) pipelines to write to Unity Catalog instead of Hive metastore.
```

Author: Not an in place update, rather it clones the pipeline and upon first update the new UC pipeline will copy over the data and streaming checkpoints.

> Upon the first update, the cloned pipeline will copy over all the data and checkpoints, and then run normally thereafter. After the cloned pipeline reaches ‘RUNNING’, both the original and the cloned pipeline can run independently.

Reviewer: Or maybe "update" is not correct due to the "clone". Could you write this out as an example:

`<original pipeline name>` -> `<original pipeline name>_cloned`?

Author: The update here refers to a step in the validation/run for the dlt pipeline. I have added clone instead of migrating to show that a new pipeline will be cloned from the original with an example.

### Known issues and Limitations:
- Only clones from HMS to UC are supported.

> - Pipelines may only be cloned into the same workspace.

Reviewer: ```suggestion
- Pipelines may only be cloned within the same workspace.
```

- HMS pipelines must currently be publishing tables to some target schema.
- Only the following streaming sources are supported:
    - Delta

>     - Autoloader

Reviewer: Link docs to autoloader

        - If your pipeline uses Autoloader with file notification events, do not run the original HMS pipeline after cloning as this will cause some file notification events to be dropped from the UC clone. If the HMS original was started accidentally, missed files can be backfilled by using the `cloudFiles.backfillInterval` option in Autoloader.
    - Kafka where `kafka.group.id` is not set
    - Kinesis where `consumerMode` is not "efo"

> - Maintenance is automatically paused (for both pipelines) while migration is in progress

Reviewer: Can you link the docs to "maintenance"?

> - If an Autoloader source specifies an explicit cloudFiles.schemaLocation, mergeSchema needs to be set to true for the HMS original and UC clone to operate concurrently.

Reviewer: ```suggestion
- If an Autoloader source specifies an explicit `cloudFiles.schemaLocation`, `mergeSchema` needs to be set to true for the HMS original and UC clone to operate concurrently.
```

The [`revert-migrated-tables` command](/docs/reference/commands#revert-migrated-tables) drops the Unity Catalog table or view and reset
the `upgraded_to` property on the source object. Use this command to allow for migrating a table or view again.

## Pipeline Migration Process

> You are required to complete the [assessment workflow](/docs/reference/workflows#assessment-workflow) before starting the pipeline migration workflow.

The pipeline migration process is a workflow that clones the Hive Metastore Delta Live Table (DLT) pipelines to the Unity Catalog.
Upon the first update, the cloned pipeline will copy over all the data and checkpoints, and then run normally thereafter. After the cloned pipeline reaches ‘RUNNING’, both the original and the cloned pipeline can run independently.

#### Example:

> Existing HMS DLT pipeline is called "dlt_pipeline", the pipeline will be stopped and renamed to "dlt_pipeline [OLD]". The new cloned pipeline will be "dlt_pipeline".

Reviewer: Yes, this is good!
